recipes/openjph/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+
+import os
+
+required_conan_version = ">=1.53.0"
+
+class OpenJPH(ConanFile):
+    name = "openjph"
+    description = "Open-source implementation of JPEG2000 Part-15 (or JPH or HTJ2K)"
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/aous72/OpenJPH"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("ht-j2k", "jpeg2000", "jp2", "openjph", "image", "multimedia", "format", "graphics")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_tiff": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_tiff": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_tiff:
+            self.requires("libtiff/4.6.0")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 14)
+
+        if self.settings.compiler == "gcc" and \
+            Version(self.settings.compiler.version) < "6.0":
+            raise ConanInvalidConfiguration(f"{self.ref} requires gcc >= 6.0")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.11 <4]")
```

Comment:
If cmake < 3.15 is needed, we don't add a tool_requires here, and instead rely on users already having them on the system (Or adding it thru profiles in the worst case scenario)

```suggestion
```